cpp/ql/lib/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -543,11 +543,11 @@ private class StdOStreamOutNonMember extends DataFlowFunction, TaintFunction {
 
   override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
     // flow from second parameter to first parameter
-    input.isParameter(1) and
+    input.isParameterDeref(1) and
     output.isParameterDeref(0)
     or
     // flow from second parameter to return value
-    input.isParameter(1) and
+    input.isParameterDeref(1) and
```

Comment:
These two assume we're using `operator<<` to output a string (not an `int`, `char` etc) but I think that's reasonable.